### PR TITLE
improvements on frameworks and IDEs

### PR DESCRIPTION
**Pull request by ewan-escience**
# Changes in this PR

This PR focuses mostly on JS frameworks and IDEs. I expect one more PR after this before issue #268 can be closed.

### CODE REVIEW

File: language_guides/javascript.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ To learn about JavaScript, view the presentations by [Douglas Crockford](https:/
 In [this video](https://www.youtube.com/watch?v=c-kav7Tf834) (47m04s), Nicholas Zakas talks about sustainability aspects, such as how to write maintainable JavaScript, how to do JavaScript testing, and good programming style (much needed in JavaScript).
 Among others, he mentions the following style guides:
 
-* [Google's style guide for JavaScript](https://google.github.io/styleguide/javascriptguide.xml);
+* [Google's style guide for JavaScript](https://google.github.io/styleguide/jsguide.html);
```
Comment (c-martinez) [inline language_guides/javascript.md:18]: At the top of the page, I see "Please note: This guide is no longer being updated. Google recommends migrating to TypeScript, and following the [TypeScript guide](https://google.github.io/styleguide/tsguide.html)." -- is it better to point to the TSguide?

File: language_guides/javascript.md [hunk 2]
```diff
@@ -29,43 +29,65 @@ Additionally, see the W3Schools page about [JavaScript best practices](https://w
...
+A UI framework that focuses on performance and being developer friendly. Like React, it uses [JSX](https://docs.solidjs.com/concepts/understanding-jsx). Read more about Solid [here](https://docs.solidjs.com/).
+
+To create a Solid application, check out the [quick start](https://docs.solidjs.com/quick-start). They also [support TypeScript](https://docs.solidjs.com/configuration/typescript).
+
+Solid has a meta-framework called [SolidStart](https://start.solidjs.com/).
 
 # JavaScript outside of the browser
 
-Most JavaScript is run in web browsers, but if you want to run it outside of a browser (e.g. as a server or to run a script locally), you'll need a JavaScript *runtime*. These are the main runtimes available:
+Most JavaScript is run in web browsers, but if you want to run it outside of a browser (e.g. as a server or to run a script locally), you'll need a JavaScript **runtime**. These are the main runtimes available:
```
Comment (c-martinez) [inline language_guides/javascript.md:90]: I wonder if in this part we could/should mention other alternatives to Node.js (e.g. write your server side script on Python). But maybe that is out of scope for the chapter.

**Review by c-martinez — APPROVED** [review]
I left a couple of questions, but even if they are not resolved, I think this could be merged.

File: language_guides/javascript.md (hunk 1 above)
Comment (ewan-escience) [reply language_guides/javascript.md:18]: Agreed, but I was planning on updating that section in the next PR.

File: language_guides/javascript.md (hunk 2 above)
Comment (ewan-escience) [reply language_guides/javascript.md:90]: I don't know about this one. On the Python page, we don't recommend other languages either...

File: language_guides/javascript.md (hunk 1 above)
Comment (c-martinez) [reply language_guides/javascript.md:18]: Perfect.

File: language_guides/javascript.md (hunk 2 above)
Comment (c-martinez) [reply language_guides/javascript.md:90]: Yeah, good point. I guess web apps is where I would use different languages for the client and server side. But we can leave it out if you think it is not super relevant.

**Comment by ewan-escience** [comment]
Can this be merged? I don't have the required permissions.
